src/metabase/util.clj

```diff
@@ -31,6 +31,16 @@
  [shared.u
   qualified-name])
 
+(defn add-period
+  "Fixes strings that don't terminate in a period."
+  [s]
+  (if (or (str/blank? s)
+          (str/ends-with? s ".")
+          (str/ends-with? s "?")
+          (str/ends-with? s "!"))
```

Comment:
```suggestion
  (if (or (str/blank? s)
          (#{\. \? \!} (last s)))
```

You can just take the last character of the string and check whether that character is in the set of `.`, `?`, or `!` instead of doing a bunch of calls to `ends-with?`

Reply:
`\x` is the syntax for a character literal in Clojure